src/shared/components/TextWithTimestamp/TextWithTimestamps.tsx

```diff
@@ -0,0 +1,59 @@
+import { convertToHtml } from '@viaa/avo2-components';
+import React, { FC, useCallback, useEffect, useRef } from 'react';
+
+import { parseDuration } from '../../helpers';
+import Html from '../Html/Html';
+
+interface TextWithTimestampsProps {
+	content: string;
+}
+
+const TIMESTAMP_REGEX = /([0-9]{2}:[0-9]{2}(:[0-9]{2})?)/g;
+
+const TextWithTimestamps: FC<TextWithTimestampsProps> = ({ content }) => {
+	const textWrapperRef = useRef<HTMLDivElement>(null);
+
+	const format = (input: string) => {
+		return input
+			.replace(/[\n\r]+/, '')
+			.replace(TIMESTAMP_REGEX, (match) => `<span class="c-timestamp">${match}</span>`);
+	};
+
+	const handleTimestampClicked = useCallback((e: MouseEvent) => {
+		const isTimestamp = (e.target as HTMLElement).classList.contains('c-timestamp');
+		const text = (e.target as HTMLElement).innerText;
+
+		if (isTimestamp) {
+			const parsed = parseDuration(text);
+			if (!isNaN(parsed)) {
+				const playingVideo: HTMLVideoElement | null = document.querySelector(
+					'.c-video-player .is-playing video'
+				) as HTMLVideoElement | null;
```

Comment:
Hoe weet deze selector welke video bedoelt wordt? Gaan we er van uit dat een gebruiker enkel een timestamp aanduidt in de beschrijving van een reeds spelende video? Wat gebeurt er als de gebruiker op een timestamp in een andere beschrijving klikt?

Reply:
we kunnen enkel video timestamp instellen op een spelende video
bij niet spelende videos is de metadata nog niet geladen, waardoor hij niet weet hoe lang de video duurt.

Het klopt dat hij nu op een timestamp kan klikken van een andere video en dat de spelende video zal verspringen.
Maar deze code werkt heel wat stabieler dan het doorgeven van meerdere refs om de timestamps te laten werken. Dus dit lijkt me wel een goeie tradeoff